native/Avalonia.Native/src/OSX/AvnView.mm

```diff
@@ -544,26 +544,15 @@ - (void)keyDown:(NSEvent *)event
     
     //InputMethod is active
     if(_parent->InputMethod->IsActive()){
```

Comment:
We can simplify this condition as well
```
if(_parent->InputMethod->IsActive())
{
    [[self inputContext] handleEvent:event];
}
else 
{
    https://github.com/AvaloniaUI/Avalonia/blob/3fb7a820a11866a54130cfa595c95867655ebc0f/native/Avalonia.Native/src/OSX/AvnView.mm#L564
```

Reply:
Yep, that's even cleaner.

Reply:
I'm trying to put this simplification into context.. are you saying to remove the handleKeyDown call? That call is necessary.

Reply:
Wrap the input context call with that condition. Raise a synthetic text input event when the condition isn't met. Always process key down.